### PR TITLE
HybridConnectionClient Request Headers feature

### PR DESCRIPTION
**Pull request by dlstucki**
## Description
Allow HybridConnectionClient to specify Custom HTTP Request Headers when connecting.

- [X] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [X] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [X] If applicable, the public code is properly documented.
- [X] Pull request includes test coverage for the included changes.
image
- [X] The code builds without any errors.

### CODE REVIEW

File: test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs [hunk 1]
```diff
@@ -153,6 +153,77 @@ async Task ConcurrentClientsTest(EndpointTestType endpointTestType)
...
+                        listenerRequestTcs.TrySetException(e);
+                        throw;
+                    }
+                };
+
+                TestUtility.Log($"Opening {listener}");
+                await listener.OpenAsync(TimeSpan.FromSeconds(30));
+
+                const string ExpectedRequestHeaderName = "CustomHeader1";
+                const string ExpectedRequestHeaderValue = "Some value here; other-value/here.";
```
Comment (bainian12345) [inline test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs]: >Some value here; other-value/here. [](start = 59, length = 34)

Does this include the invalid values?

Comment (dlstucki) [reply test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs]: No, this value is valid.

---
In reply to: [266133306](https://github.com/Azure/azure-relay-dotnet/pull/97#discussion_r266133306) [](ancestors = 266133306)

Comment (bainian12345) [reply test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs]: I don't know if it's worthwhile to explictly test for invalid values

---
In reply to: [266133760](https://github.com/Azure/azure-relay-dotnet/pull/97#discussion_r266133760) [](ancestors = 266133760,266133306)

Comment (dlstucki) [reply test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs]: added in the 3rd iteration.

---
In reply to: [266134567](https://github.com/Azure/azure-relay-dotnet/pull/97#discussion_r266134567) [](ancestors = 266134567,266133760,266133306)

**Pull request merged**